Sources/Interpreter/Interpreter.swift

```diff
@@ -10,48 +10,32 @@ struct CodePointer {
 
 }
 
+/// The value produced by executing an instruction.
+typealias InstructionResult = Any
+
+/// Address to allocated object with type layout.
+struct Address {
+
+  /// Address of object in memory.
+  public let memoryAddress: Memory.Address
+
+  /// Type layout of object.
+  public let memoryLayout: TypeLayout
```

Comment:
I think the role of this is that it's the type part of the typed location.  The whole layout is stored merely as a convenience, but that could instead be looked up at any time.


```suggestion
  /// The type to be accessed at `position`.
  public let type: TypeLayout
```